src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Cache.Memory;
+using Microsoft.Framework.FileSystemGlobbing;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
+{
+    /// <summary>
+    /// Utility methods for <see cref="ITagHelper"/>'s that support attributes containing file globbing patterns.
+    /// </summary>
+    public class GlobbingUrlBuilder
+    {
+        private static readonly char[] PatternSeparator = new[] { ',' };
+
+        private readonly FileProviderGlobbingDirectory _baseGlobbingDirectory;
+
+        // Internal for testing
+        internal GlobbingUrlBuilder() { }
+
+        public GlobbingUrlBuilder([NotNull] IFileProvider fileProvider, IMemoryCache cache, PathString requestPathBase)
```

Comment:
nit: not sure why you're bothering w/ XML comments for a class in this namespace (low-priority as you've seen with the lack of complaints about `FileProviderGlobbingDirectory` et cetera).  but, if you're going to document, document all `public` (and `protected` members), including constructors.